yal/assessments.py

```diff
@@ -214,17 +238,210 @@ async def state_assessment_later(self):
                     "",
                     "Check you later 🤙🏾",
                 ]
             )
         )
+        # TODO: set the reminder type to later 1hour
         return WhatsAppButtonState(
             self,
             question=question,
             choices=[Choice("menu", "Go to main menu")],
             error=get_generic_error(),
-            next=MainMenuApplication.START_STATE,
+            next="state_pre_mainmenu",
+        )
+
+    async def state_handle_assessment_reminder_response(self):
+        inbound = utils.clean_inbound(self.inbound.content)
+        if inbound in ["continue now", "lets do it", "ask away", "start the questions"]:
+            msisdn = utils.normalise_phonenumber(self.inbound.from_addr)
+            whatsapp_id = msisdn.lstrip(" + ")
+            data = {
+                "assessment_reminder_sent": "",  # Reset the field
+            }
+
+            error = await rapidpro.update_profile(whatsapp_id, data, self.user.metadata)
+            if error:
+                return await self.go_to_state("state_error")
+            return await self.go_to_state(
+                f"state_{self.user.metadata['assessment_reminder_name']}_assessment"
+            )
+
+        if inbound == "skip":
+            return await self.go_to_state("state_stop_assessment_reminders_confirm")
+
+        if inbound == "remind me in 1 hour":
+            self.save_metadata("assessment_reminder_hours", "1hour")
+            return await self.go_to_state("state_reschedule_assessment_reminder")
+
+        if inbound == "remind me tomorrow":
+            self.save_metadata("assessment_reminder_hours", "23hours")
+            return await self.go_to_state("state_reschedule_assessment_reminder")
+
+    async def state_stop_assessment_reminders_confirm(self):
+        assessment_reminder_name = self.user.metadata["assessment_reminder_name"]
+
+        if assessment_reminder_name == "locus_of_control":
+            return WhatsAppButtonState(
+                self,
+                question=self._(
+                    "\n".join(
+                        [
+                            "Please take note👆🏽 you can't access all parts of the "
+                            "Bwise bot if you don't complete the questions first.",
+                            "",
+                            "You can still use the menu and ask questions, but I "
+                            "can't give you a personalised journey.",
+                            "",
+                            "*Are you sure you want to skip?*",
+                        ]
+                    )
+                ),
+                choices=[
+                    Choice("skip", self._("Yes, skip it")),
+                    Choice("start", self._("Start questions")),
+                ],
+                next={
+                    "skip": "state_stop_assessment_reminders",
+                    "start": assessment_reminder_name,
```

Comment:
Shouldn't this be something like `f"state_{assessment_reminder_name}_assessment"`? Maybe we need a test for this user path as well?